homeassistant/components/buienradar/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for buienradar2 integration."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_LATITUDE, CONF_LONGITUDE, CONF_NAME
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+
+from .const import (
+    CONF_CAMERA,
+    CONF_COUNTRY,
+    CONF_SENSOR,
+    CONF_WEATHER,
+    DOMAIN,
+    HOME_LOCATION_NAME,
+    SUPPORTED_COUNTRY_CODES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return a set of configured SimpliSafe instances."""
+    entries = []
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        entries.append(
+            f"{entry.data.get(CONF_LATITUDE)}-{entry.data.get(CONF_LONGITUDE)}"
+        )
+    return set(entries)
+
+
+class BuienradarFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for buienradar2."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Init MetFlowHandler."""
+        self._errors = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        self._errors = {}
+
+        if user_input is not None:
+            if (
+                f"{user_input.get(CONF_LATITUDE)}-{user_input.get(CONF_LONGITUDE)}"
+                not in configured_instances(self.hass)
+            ):
+                name = user_input[CONF_NAME]
+
+                data = {
+                    CONF_NAME: name,
+                    CONF_LATITUDE: user_input[CONF_LATITUDE],
+                    CONF_LONGITUDE: user_input[CONF_LONGITUDE],
+                    CONF_CAMERA: user_input[CONF_CAMERA],
+                    CONF_SENSOR: True,
+                    CONF_WEATHER: True,
+                    CONF_COUNTRY: user_input[CONF_COUNTRY],
+                }
+
+                return self.async_create_entry(title=name, data=data)
```

Comment:
I would extract this to a separate function so the logic is easier to follow here.